File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int calc_index(int letters, int words, int sentences);

int main(void)
{
    string text = get_string("Text: ");
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);
    int index = calc_index(letters, words, sentences);


}








int calc_index(int letters, int words, int sentences)
{
    float L = (letters * 100) / words;
    float S = (sentences * 100) / words;

    return (int) (0.0588 * L - 0.296 * S - 15.8)
}




















int count_sentences(string text)
{
     int sentences = 0;
     for (int i = 0, n = strlen(text); i < n; i++)
     {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
     }
     return sentences;
}












int count_words(string text)
{
     int words = 1;
     for (int i = 0, n = strlen(text); i < n; i++)
     {
        if (text[i] == ' ')
        {
            words++;
        }
     }
     return words;
}













int count_letters(string text)
{
     int letters = 0;
     for (int i = 0, n = strlen(text); i < n; i++)
     {
        if (isalpha(text[i]))
        {
            letters++;
        }
     }
     return letters;
}

File: sentimental-readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)
    index = calc_index(letters, words, sentences)
    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


def calc_index(letters, words, sentences):
    L = (letters * 100) / words
    S = (sentences * 100) / words
    index = 0.0588 * L - 0.296 * S - 15.8
    return round(index)


def count_letters(s):
    letters = 0
    for i in s:
        if i.isalpha():
            letters += 1
    return letters
    

def count_words(s):
    words = 1
    for i in s:
        if i == ' ':
            words += 1
    return words


def count_sentences(s):
    sentences = 0
    for i in s:
        if (i == '.' or i == '?' or i == '!'):
            sentences += 1
    return sentences


main()